### PR TITLE
Add staking/leasing transaction support for waves

### PR DESCRIPTION
**Pull request by vladislavpetushkov**
* change protobuf messages for Waves to support diff type of transactions

* rewrite transactions class to suport diff type of transactions

* rewrite tests to suport diff type of transactions

* rewrite interface for to suport diff type of transactions

* add test for staking/leasing transactions

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

**Comment by leoneparise** [comment]
Nice! I'll take a look

**Comment by leoneparise** [comment]
@hewigovens How do we solve codecov issues?

File: src/Waves/Transaction.cpp [hunk 1]
```diff
@@ -17,27 +17,29 @@ using json = nlohmann::json;
...
+        fee_asset = Transaction::WAVES;
     }
+    
     data.resize(2);
     data[0] = static_cast<byte>(TransactionType::transfer);
     data[1] = static_cast<byte>(TransactionVersion::V2);
     append(data, pub_key);
-
-    if (asset == WAVES) {
+
```
Comment (hewigovens) [inline src/Waves/Transaction.cpp]: extra empty spaces

**Comment by hewigovens** [comment]
@leoneparise @vladislavpetushkov codecov failed because `jsonLease` and `jsonCancelLease` are not covered in `Transaction.cpp`

**Comment by hewigovens** [comment]
still need a testcase to cover `jsonCancelLease`

**Pull request merged**